REPOSITORY,ISSUE_ID,STATUS,EVENT,CREATED_AT,TIME DURATION,NEGATIVE,NEUTRAL,POSITIVE,MSG,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,opened,2011-04-02 06:45:47,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,title,2011-04-02 06:45:47,00:00,,1,,"b""b' Python 3 fixes'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,body,2011-04-02 06:45:47,00:00,,1,,"b""b' This makes several more things work with Python 3"," including:authentication on HTTPS connectionsRoute 53S3 get_bucket and bucket operations'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,1,2011-04-02 06:45:47,16.12,,1,,"b'b\' There are still a couple of py3-incompatible uses of unicode in  ( even has a SyntaxError from u"""")", which I haven\\\'t addressed because I don\\\'t use SDB and am not familiar with the intended semantics.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,closed,2011-04-02 07:01:54,16.12,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,reopened,2011-04-02 07:02:02,0.13,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,2,2011-04-02 06:45:47,0.33,,1,,"b'b"" (didn\'t mean to close"," whoops)""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,3,2011-04-02 06:45:47,888.88,,,1,"b""b' I have merged these changes to the branch_master_python3k branch.  thanks!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,closed,2011-04-02 21:51:07,889.08,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,151,closed,-,-,-,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,opened,2011-05-05 05:56:29,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,title,2011-05-05 05:56:29,00:00,,1,,"b""b' Gainfully using mock_storage_'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,body,2011-05-05 05:56:29,00:00,1,,,"b'b"" I don\'t think it\'s documented"," but it\'s included and terribly useful. I made it slightly more useful with only a few minor tweaks.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,1,2011-05-05 05:56:29,488.87,,,1,"b'b"" Thanks for updating the mock_storage_I originally built it for the gsutil unit tests and I\'m glad others are finding it useful.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,closed,2011-05-05 14:05:21,488.87,1,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,reopened,2011-05-05 14:06:24,1.05,1,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,closed,2011-05-05 14:07:07,0.72,1,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,2,2011-05-05 05:56:29,940.45,,,1,"b""b' Sure!  I just did another minor change (f6bac55) to remove the  assumption of a single mock.  Makes for much better fixture  repeatability (without having to make a tearDown() that disposes of  the global mock.) On May 5",2011, at 7.05 ," mfschwartz wrote:'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,183,closed,-,-,-,1,1,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,opened,2011-07-18 19:28:20,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,title,2011-07-18 19:28:20,00:00,,1,,"b'b"" RDS call modify_dbinstance with multi_az = True doesn\'t actually set an instance to MultiAZ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,body,2011-07-18 19:28:20,00:00,,,1,"b'b"" I didn\'t have permission to reopen #74", but this issue is still present. The weird thing is that it the request boto sends looks perfectly legit as far as I can see. We get a 200 status code and then... nothing happens. It doesn\'t show as a pending modified value and no MultiAZ copy is ever created.  Debugging output: From the documentation," that seems like exactly what I\'d expect. Is there any chance this is an Amazon API bug?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,1,2011-07-18 19:28:20,308.65,,,1,"b'b"" Unsure whether this is related"," but amazon doesn\'t even seem to tell us whether multi-az is enabled in the data returned by the Action=DescribeDBInstances call.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,2,2011-07-18 19:28:20,829.45,,1,,"b""b' It looks like the DescribeDBInstances response has a multiaz element: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,closed,2011-10-18 14:26:26,1138.1,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,reopened,2011-10-18 14:26:49,0.38,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,3,2011-07-18 19:28:20,109.65,,1,,"b'b"" After a bit of experimentation"," I discovered that the lack of multiaz in the response seems fixed when I change this line: to reflect the 2011-04-01 mentioned in winhamwr\'s example.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,4,2011-07-18 19:28:20,1.42,1,,,"b'b"" Sorry", not in winhamwr\'s example," but in the docs mentioned by garnaat.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,5,2011-07-18 19:28:20,1190.0,,1,,"b""b' I believe this has been fixed in afcf018e'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,closed,2011-10-25 12:07:30,1300.68,1,4,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,265,closed,-,-,-,1,4,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,opened,2011-07-26 13:34:45,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,title,2011-07-26 13:34:45,00:00,,1,,"b""b' 505 HTTP Version Not Supported'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,body,2011-07-26 13:34:45,00:00,,1,,"b""b' Occasionally getting a 505 with CloudFormation requests. Is this just an AWS issue? Some query params omitted for privacy:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,1,2011-07-26 13:34:45,134.4,,1,,"b'b"" Seems to happen mostly when I\'m doing stuff in the AWS Web Console simultaneously. I don\'t think this is a Boto issue""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,closed,2011-07-26 15:49:09,134.4,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,reopened,2011-07-26 16:57:13,68.07,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,2,2011-07-26 13:34:45,1412.9,,1,,"b""b' I believe this should be closed.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,closed,2011-07-27 15:22:03,1344.83,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,3,2011-07-26 13:34:45,1370.13,,1,,"b'b"" I\'ve started a thread on AWS forums regarding this error. The inconsistency makes me think it has to do with which API server the request gets routed to. Here\'s the thread: url ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,279,closed,-,-,-,0,5,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,opened,2011-11-08 21:38:49,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,title,2011-11-08 21:38:49,00:00,,1,,"b""b' AutoScaleConnection constructor sets debug param to None rather than 0'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,body,2011-11-08 21:38:49,00:00,,1,,"b""b' This causes a problem if you try to create an AutoScaleConnection without specify a value for debug.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,closed,2011-11-08 21:40:47,1.97,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,1,2011-11-08 21:38:49,687.4,,1,,"b'b"" I\'m having the same issue with v2.1.1.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,2,2011-11-08 21:38:49,254.18,,1,,"b'b"" I\'m unable to reproduce this.  Could you give me a little more info about what you are doing and the error you are receiving?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,3,2011-11-08 21:38:49,321.55,,,1,"b""b'   With the following ~/.boto: I get a TypeError and this traceback: But everything works fine with the [Boto] section in my ~/.boto.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,reopened,2012-01-31 19:01:31,1280.73,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,closed,2012-01-31 19:01:51,0.33,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,393,closed,-,-,-,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,opened,2011-12-02 01:56:42,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,title,2011-12-02 01:56:42,00:00,,1,,"b""b' Cant use dot for bucketname '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,body,2011-12-02 01:56:42,00:00,,1,,b'b\' code is here : from  import S3Connectionconn =S3Connection(\\\'XXXXX\\\',\\\'XXXXX\\\')rs=()for b in rs:  print b=(\\\'AAAAA\\\')  for key in (): -## 1 I create two bucket ( photo_bucket ," ) when I get the bucket name (photo_bucket) have no char \\\'.\\\' in it .it works . but when I use """" ",it has error . InvalidCertificateException: Host  returned an invalid certificate (hostname mismatch): {\\\'notAfter\\\': \\\'Dec 18 23:59:59 2013 GMT\\\', \\\'subjectAltName\\\': ((\\\'DNS\\\', \\\'_.s3\\\'), (\\\'DNS\\\', \\\'s3\\\')), \\\'subject\\\': (((\\\'countryName\\\', u\\\'US\\\'),), ((\\\'stateOrProvinceName\\\', u\\\'Washington\\\'),), ((\\\'localityName\\\', u\\\'Seattle\\\'),), ((\\\'organizationName\\\', u\\\' Inc.\\\'),), ((\\\'commonName\\\', u\\\'_.s3\\\'),))} can anyone fix it ? \'',,,,,,,,,,,,,,,,,,,
boto,421,closed,1,2011-12-02 01:56:42,686.53,,1,,"b'b"" I don\'t understand the error you are receiving.  I am unable to reproduce this:  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,2,2011-12-02 01:56:42,37.07,,,1,b'b\' thanks for you reply .I download the newest boto .and test it . My system is windows 7.I have set cnames :  as  s3  here is my code: ######## import sysimport globimport osfrom  import S3Connectionfrom  import Locationfrom  import Key conn =S3Connection(\\\'xxxxxx\\\',"\\\'xxxxxx\\\') # xxxx is my private amazon aceess code .b = (\\\' \\\')  # here is my bucket name for k in b :    print  ##########idle generate it:Traceback (most recent call last):  File ""D:/""", line 38," in   for k in b :also I use :b=(\\\'s3\\\')  it give:: Host s3 returned an invalid certificate (remote hostname ""s3"" does not match certificate): {\\\'notAfter\\\': \\\'Dec 18 23:59:59 2013 GMT\\\'", \\\'subjectAltName\\\': ((\\\'DNS\\\', \\\'_.s3\\\'), (\\\'DNS\\\', \\\'s3\\\')), \\\'subject\\\': (((\\\'countryName\\\', u\\\'US\\\'),), ((\\\'stateOrProvinceName\\\', u\\\'Washington\\\'),), ((\\\'localityName\\\', u\\\'Seattle\\\'),), ((\\\'organizationName\\\', u\\\' Inc.\\\'),), ((\\\'commonName\\\', u\\\'_.s3\\\'),"))} also I create new  bucket name :it give the same error :: Host  returned an invalid certificate (remote hostname ""3"" does not match certificate): {\\\'notAfter\\\': \\\'Dec 18 23:59:59 2013 GMT\\\'", \\\'subjectAltName\\\': ((\\\'DNS\\\', \\\'_.s3\\\'), (\\\'DNS\\\', \\\'s3.amazonaws.com\\\')), \\\'subject\\\': (((\\\'countryName\\\', u\\\'US\\\'),), ((\\\'stateOrProvinceName\\\', u\\\'Washington\\\'),), ((\\\'localityName\\\', u\\\'Seattle\\\'),), ((\\\'organizationName\\\', u\\\' Inc.\\\'),), ((\\\'commonName\\\', u\\\'_.s3\\\'),))}Does it need to config in amazon s3 ? or where do to it?\''
boto,421,closed,3,2011-12-02 01:56:42,215.88,,1,,"b'b"" I can confirm this behavior. b is None and get_bucket returns . Is there a way to disable certificate checking? Because with a dot in the name"," the url pattern on the SSL certificate won\'t match.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,4,2011-12-02 01:56:42,175.05,,1,,"b'b"" Seeing this as well. Perhaps there\'s a way to force boto to use HTTP?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,5,2011-12-02 01:56:42,3.85,,,1,"b""b' Found it. My ~/.boto configuration file had the line: By changing the value to False"," I resolved my issue. Hope this helps!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,closed,2013-07-11 23:06:04,1269.37,0,5,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,reopened,2013-07-11 23:07:07,1.05,0,5,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,6,2011-12-02 01:56:42,153.6,,1,,"b""b' You can also change the calling format via the arg to not use the instead of disabling SSL.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,closed,2013-07-11 23:08:41,1.57,0,6,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,7,2011-12-02 01:56:42,1355.0,,1,,"b""b' This worked for me:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,8,2011-12-02 01:56:42,1360.28,,1,,"b'b"" I\'m seeing this using the bucket name via [yas3fs]url  Because I\'m not in direct control of my boto config file", I do not believe I can disable certificate checking. Nonetheless," that seems an unappealing fix regardless. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,421,closed,-,-,-,0,8,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,opened,2012-01-17 05:21:20,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,title,2012-01-17 05:21:20,00:00,,1,,"b""b' slick53 methods grafted onto to boto route53 connection'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,body,2012-01-17 05:21:20,00:00,,1,,"b'b"" As mentioned on the boto email list", I\'d like to see slick53\'s more intuitive methods available in the boto package.  Aside from minor changes to make the code more idiomatic, I\'ve basically left the code alone, though I may revisit it pending review of this request. There are some basic unit tests. If accepted," I\'ll submit some changes to the docs.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,1,2012-01-17 05:21:20,536.75,,1,,"b'b"" I\'d prefer if we put slick53 into a  module to keep the connection modules as close to the API as possible. This would follow what the Django project does some of the most useful Django modules such as the Admin interface is included not in core but in contrib. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,2,2012-01-17 05:21:20,36.63,,,1,"b'b"" Fine by me.  I wasn\'t aware boto had a contrib directory.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,closed,2012-01-17 14:56:06,574.77,0,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,3,2012-01-17 05:21:20,5.98,,,1,"b'b"" Might want to make sure this is what @garnaat had in mind (putting this in contrib). The route53 API we currently expose is very", very quirky and has some usability issues. I tinkered with @rubic\'s original slick53, and was pretty impressed with the improvement seen over the standard  module. Also, we pretty commonly inter-mix Amazon API methods and convenience methods in our other sub-modules. This is effectively what a lot of slick53  be. Convenience methods that fill in the gaps that the standard API leaves as an exercise for the user. @garnaat asked for @rubic to bring this over," maybe he could pipe in?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,reopened,2012-01-17 15:22:17,26.18,0,3,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,4,2012-01-17 05:21:20,28.6,,,1,"b'b"" My original post was similar to gtaylor\'s thoughts", but I didn\'t want to intrude.  Generally I agree most enhancements should be handled in a contrib directory. However in the case of route53, the boto code -while following the semantics of AWS -is almost unusable, IMO.  I\'m still happy to go either direction. Also," Brad Carleton is the author of slick53.  I\'m just interested in getting it into boto.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,5,2012-01-17 05:21:20,43.42,,1,,"b""b' I agree that this should be a core part of boto rather than in contrib.  I have been experimenting with a layered approach for another module and I think that works pretty well.  I can take a crack at doing something similar here. Mitch On Jan 17",2012, at 10:29 AM," Jeff Bauerreply@ wrote:'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,6,2012-01-17 05:21:20,558.23,,,1,"b""b' Hi all"," I just wanted to say I am excited that this is getting moved into boto! And thank you Jeff for merging the code.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,7,2012-01-17 05:21:20,252.5,,,1,"b'b"" I agree with using a layered approach.  My concern with the commit in this pull request is it doesn\'t appear to handle Amazon\'s aliasing or WRRs.  While the convenience methods are probably better for the majority of the users of boto"," in their current form I\'m guessing (haven\'t tested) that they will break in unexpected ways when they encounter an alias RR and know they will break when they encounter a WRR. I\'ve been trying to convince an unnamed author for two months to submit a pull request for his WRR patch.  Sometimes I feel that I may be the only person using boto\'s route53 code with ELB. :(  I\'m happy to put code where my mouth is to ensure the final form properly handles alias RRs and WRRs.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,8,2012-01-17 05:21:20,19.22,,1,,"b""b' @jimbrowne:  Would your patches for RR", WRR affect the proposed method signatures for the proposed code?  In my mind, that would be the main issue. Not having looked into the  situation I may be talking out of my ass, but could we just make the functions _fail_ if those situations are encountered (until patched)," which would admittably affect a minority of users?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,9,2012-01-17 05:21:20,30.02,,1,,"b'b"" I think the fundamental problem is most people see DNS RRs as being a tuple of (type", value, ttl).  In Route53, they are actually (type, value, ttl, wrr set).  The implicit goal of the slick53 API, as I see it, is to simplify the interface.  One could reasonably argue that most will never use Route 53 ALIAS records nor WRR so the simplified API should ignore them. However, there are tools out there that generate WRRs and ALIAS records, the most likely to be encountered are the elb-(dis)associate CLI commands.  For some reason they create the ALIASes as WRRs.  That\'s how I stumbled upon this problem in the first place; I was moving a system formerly controlled by the ELB CLI to boto-based code that manipulated route53 and elb.  Another possible source of records with WRRs is the AWS Console as it now supports direct manipulation of Route53 RRs. If one ignores WRRs it breaks being liberal in what you accept and this problem will trip up anyone who encounters pre-existing WRRs.  The functions will fail as the changes will fail as Route53 will not be able to match the (type, value, ttl) tuple to the existing (type, value, ttl, wrr set) tuple. At the minimum I want to see WRR support incorporated into the lower level.  Perhaps the right approach is to just put a prominent warning in the documentation on the slick53 derived methods noting that they do not support WRRs and encountering WRRs will cause errors.  Those who need to manipulate records with WRRs can use the lower level API. I just wanted to raise this aspect for discussion. Disclaimer: I was last knee deep in Route53 about two months ago," so my description might be slightly inaccurate.""'",,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,10,2012-01-17 05:21:20,3.93,,,1,"b'b"" I think we can safely say that the lower level API should support WRRs", as the lower level API should closely mirror the AWS API. Work on that could start now without question, this is the way we\'ve been handling a few of the more recently developed sub-modules. After that," perhaps we can see how the chips fall with the higher level API?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,11,2012-01-17 05:21:20,426.15,,1,,"b'b\' @jimbrowne Taking into account what you say ""most people see DNS RR\\\'s as being a tuple of (type", value," ttl)""", I don\\\'t think that the WRR\\\'s really break that concept.  It just changes the meaning of value.  In this case value could be a regular python dictionary specifying the weights, that way the same function could support both.  And this would be super easy for the end user. I also don\\\'t see how (type, value, ttl) does not apply for the AWS Alias records as well, although I am not really familiar with them. In short, I think (type, value, ttl) is the right abstraction for DNS records.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,12,2012-01-17 05:21:20,288.18,,1,,"b'b"" I second gtaylor\'s conclusion and bluepines\' suggestion regarding dictionary values.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,13,2012-01-17 05:21:20,883.13,,1,,b'b\' I had some time finally to look at this today.  The issue is the tuple is really (name, type, wrr_id, wrr_weight) with many subtleties.  The value doesn\\\'t matter as you can not have more than one (name, type) without using WRRs.  Also, the ALIAS type doesn\\\'t require a TTL as it inherits the TTL from the backing ELB.  Confused yet? :) Here\\\'s a tricky example:Case 1, a single RR with multiple values, basically round robin DNS.Case 2, two WRRs with the same name, type and different (wrr_id, wrr_weight). Would the idea be that the value dictionary is constructed like: {\\\'1.2.3.4\\\': 20, \\\'5.6.7.8\\\': 40} and boto would build the two WRRs needed if the weights differ or would build a single RR with a multiple values when the weights are equal?  SetIdentifiers would have to be generated, but I believe they are scoped to (name," type) so that shouldn\\\'t be a problem (e.g. use ""id1""",id2, ...). If that is what you mean I like the idea as it hides WRRs completely while enabling their use and has boto deal with all of the ugliness behind the scenes. I\\\'m happy to code this up if it sounds correct, as well as adding the ALIAS case. (I just submitted a pull request for WRR support overall: url \'',,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,14,2012-01-17 05:21:20,262.8,,,1,"b""b' +1 and thanks for your effort Jim.  You seem to be the boto user most current with the WRR issues and your pull request is backward compatible with the current api.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,15,2012-01-17 05:21:20,314.88,1,,,"b'b"" An oldie but a goodie.  It seems a shame to waste all of this good discussion so let\'s revive this and try to get it merged.  I know @rubic is still interested.  How about @jimbrowne ?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,16,2012-01-17 05:21:20,233.78,,1,,"b'b"" I\'ll take a look at it tonight.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,17,2012-01-17 05:21:20,570.52,,1,,"b""b' While re-familiarizing myself with this found and opened url First thoughts tonight after a quick reading are to split the zone changes from the record changes and accept the former immediately while reviewing the latter; though I should be able to finish up something for all of this request this week.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,18,2012-01-17 05:21:20,1208.25,1,,,b'b\' Questions (mainly for @garnaat and @bluepines): First: The API is inconsistent . MX records; the MX related calls only manipulate records attached to the zone top.  IOW one can\\\'t create MX records for anything but the zone.  I think the MX calls should accept a name parameter. Second: Since the Zone object already contains the DNS zone name," would it be better if the methods operating on zones automatically post-pend the zone name to the name parameter as a convenience rather than just ensuring the name parameter is fully qualified (i.e. ends in ""."")?  I note that the repr for records returned by get_records will be fully qualified (i.e. include the zone name). For example this: would instead be this: but that raises the issue of how to handle the zone top.  Pass \\\'\\\' as the name parameter?  I just find including the zone name on every name attribute to be repeating oneself and would probably result in code like: . WRR and LBR I\\\'ll try out some changes tonight.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,19,2012-01-17 05:21:20,197.6,,1,,"b'b"" I\'ve merged the proposed patch against the current develop branch and split out the import fix-ups and **repr** changes that were formally monkey-patched.  Still looking at the rest of the code and its interaction with WRR and LBR.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,20,2012-01-17 05:21:20,1416.6,,1,,"b'b"" I\'ve re-factored the original code quite a bit and have added support for WRR and LBR; I\'m still running through some tests before pushing. I think I might have found a bug in the Route53 API as well. Should have something to push tomorrow night.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,21,2012-01-17 05:21:20,1006.38,,,1,"b""b' Awesome news Jim!  Thanks for making the effort.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,22,2012-01-17 05:21:20,482.82,1,,,"b'b"" Code is finished", additional test cases have been written, and all tests pass.  I just want to add some documentation and will push. Oh, and I didn\'t find a bug in the Route53 API; I just had forgotten that passing a  to get_all_records just sets a starting point... it\'s not a true filter.  Not sure why they built it that way," but the service works per the documentation.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,23,2012-01-17 05:21:20,1361.38,,1,,"b""b' The above commit completes the work I wanted to propose for a new Layer 2.  Would appreciate review by those on this issue to guide the hand of @garnaat'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,24,2012-01-17 05:21:20,1226.37,,,1,"b""b' Jim: It looks good to me.  @garnaat: Can this be put on track for 2.4?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,25,2012-01-17 05:21:20,1075.65,1,,,"b""b' Sorry for dragging this out even longer. The code looks good.  Could use more docstrings.  In looking through this"," it seems like the best thing to do would be to leave the current  unchanged and to add a layer2.py file that contains all of the slick53 code that was added to .  We could then leave the default behavior (i.e. connect_route53()) to still use the original but then people could still create a Layer2 connection. Does that make sense?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,26,2012-01-17 05:21:20,1251.92,,1,,"b""b' @garnaat:  Mitch", the resulting merge will not require any additional imports," correct?  How will the connection differ from connect_route53() if the user wants to use the slick53 methods?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,27,2012-01-17 05:21:20,701.48,,1,,"b'b"" @garnaat did you look at the commits I suggested"," or just the original pull request?  I ask because I added docstrings to every function in my version.  Would it make more sense for me to just open a separate pull request for ? For my commit I could move the Zone and Status objects to separate files if you\'d like.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,28,2012-01-17 05:21:20,688.58,,,1,"b'b"" The code looks great.  Thanks. My only question was on organization of the code.  Some modules like dynamodb have a layer1.py and layer2.py where layer1 is low-level", close to the protocol and layer2 provides a higher-level, more pythonic interface.  We could package this in a similar way, but it\'s not a requirement from my point of view.  If you feel this organization makes the most sense," I\'m fine with it.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,29,2012-01-17 05:21:20,1130.83,,,1,"b""b' I was just looking for a pythonic wrapper around boto route53 and this looks like exactly it. Would love to see this merged (or those changes backported into slick53 as a standalone module).'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,30,2012-01-17 05:21:20,1227.37,,1,,"b""b' @garnaat I changed the code layout", renamed the test to not mention layer2," and fixed an issue noted by @gingerlime.  I think my branch is finally ready for merging.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,closed,2012-12-14 18:14:43,172.43,4,17,11,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,478,closed,-,-,-,4,17,11,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,opened,2012-01-23 10:17:00,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,title,2012-01-23 10:17:00,00:00,,1,,"b""b' Added missing (boolean) is_latest attribute to Key object.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,body,2012-01-23 10:17:00,00:00,,1,,"b""b' added missing attribute is_latest to key object  default value is False.is_latest is set when reading XML response of get versions  in the same way the DeleteMarker is_latest gets set.Added unit tests to test_'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,1,2012-01-23 10:17:00,3.2,,1,,"b'b\' Also note: I\\\'d like to pull mfa stuff out of the test_ at some stage into a perhaps test_ as I think its useful to be able to run the test_versioning stuff separately. I don\\\'t have an mfa key myself so I have to disable the whole test right now when I want to test stuff. then just have test suites for ""s3"" and ""s3nomfa"" instead of ""s3nover"". Can do that on another branch though once this is pulled in if people like the idea.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,2,2012-01-23 10:17:00,374.52,,1,,"b""b' Yes"," I agree with you on splitting out versioning and MFA.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,closed,2012-01-25 16:34:43,377.72,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,reopened,2012-01-25 16:35:36,0.88,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,3,2012-01-23 10:17:00,1.48,1,,,b'b\' Sorry," just hit the ""close & comment"" button rather than ""comment"".  I\\\'m about to merge this so it will then be really closed.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,closed,2012-01-25 16:36:19,0.72,1,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,510,closed,-,-,-,1,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,opened,2012-02-03 16:17:48,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,title,2012-02-03 16:17:48,00:00,,1,,"b""b' The limit parameter is ignored in Layer2.query()'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,body,2012-02-03 16:17:48,00:00,,1,,"b""b' If you pass a limit parameter to Layer2.query()"," it will be used to limit the number of results returned in each request but the generator will still loop over all available results before returning.  This is almost certainly not what the caller had in mind.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,1,2012-02-03 16:17:48,17.85,,1,,"b""b' I disagree", I think this is exactly the type of functionality it needs to perform. Specifically because if you do limit each page to a certain number of results, then you can help to throttle your total usage per API request," instead of using a burst of Read units all at once. See url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,2,2012-02-03 16:17:48,9.97,,1,,"b""b' But the net result is that there is no way to actually limit the number of results you get.  You will always get all of them"," the only thing you are controlling is how many are requested from DynamoDB at a time.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,3,2012-02-03 16:17:48,58.62,,1,,"b'b\' Maybe the right answer is ""both"".  In other words"," maybe we want to be able to control the number of results requested at the API level and also be able to control the total number of results returned and those are really two different things.  I think we are overloading the definition of ""limit"" right now.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,4,2012-02-03 16:17:48,18.28,,1,,b'b\' agreed," what if we have a ""limit"" and a ""page_limit"" ?\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,closed,2012-02-03 18:02:31,104.72,0,6,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,reopened,2012-02-03 18:02:35,0.07,0,6,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,5,2012-02-03 16:17:48,0.25,1,,,"b""b' sorry"," that close button is way too close to the comment button..'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,6,2012-02-03 16:17:48,6.83,,1,,b'b\' Yeah," I\\\'ve done that more than once. I can\\\'t tell which is which between ""limit"" and ""page_limit"".  How about request_limit to specify the number of items in the request (i.e. the Limit param) and either ""limit"" or ""max_results"" for the max number of items you want to get back.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,7,2012-02-03 16:17:48,3.8,,1,,"b""b' I think changing both would be the best"," so request_limit and max_results. This should be done the same in the scan function too.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,8,2012-02-03 16:17:48,7.22,,1,,"b""b' Agreed.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,closed,2012-02-06 19:45:12,102.62,1,9,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,549,closed,-,-,-,1,9,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,opened,2012-02-06 14:21:22,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,title,2012-02-06 14:21:22,00:00,,1,,"b""b' [EC2]  method'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,body,2012-02-06 14:21:22,00:00,,1,,"b""b' The documentation describes the method ", seeurl  There is no such a method in the library (I am using boto 2.1.1," downloaded from here url  even though I can see it hereurl '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,1,2012-02-06 14:21:22,110.85,,1,,"b""b' Hi The latest release of boto is 2.2.1 and is available from url  and from the github download tab.  I neglected to upload the latest version to Google Code.  I will do that now. Mitch'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,closed,2012-02-06 16:12:13,110.85,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,2,2012-02-06 14:21:22,19.0,,1,,"b'b"" Hello Mitch", I have tried version 2.2.1 from github, and I can see that method in the source code. However the method is not available in the installed library (I have issues the install as well as build method -a few **init**.py files are missing," so the original tar file won\'t build).""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,reopened,2012-02-06 17:33:36,81.38,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,3,2012-02-06 14:21:22,66.57,,1,,"b'b\' Hmm.  I just tried downloading the boto-2.2.1 file from PyPI and did a ""python  install"" in a clean virtualenv and it seemed to work without any problem and I confirmed that the get_all_instance_status method was there. Could you provide a bit more info?\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,4,2012-02-06 14:21:22,1033.95,,,1,"b""b' Hello Mitch", I have downloaded the tar file once again (2.2.1) and installed it in a clean environment (python 2.7).I can confirm that it works now. Thank you,"Michele '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,closed,2012-02-07 10:51:44,1038.13,0,5,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,559,closed,-,-,-,0,5,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,opened,2012-02-07 23:45:48,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,title,2012-02-07 23:45:48,00:00,,1,,"b""b' Boto connection timeout'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,body,2012-02-07 23:45:48,00:00,1,,,"b'b\' My program uses SQS a lot over the course of many days. I have noticed that my program will fall to 0% cpu usage and just get stuck their for a long time. When the program is suck i can debug it to obtain the following stack trace.  It appears as though the socket isn\\\'t timing out properly.  I think the best solution is to expose httplib\\\'s timeout value on Boto\\\'s connection object so that I can lower it.  What does the Boto community think?   File ""/""", line 245," in get_messages  attributes=attributes) File ""/""", line 157, in receive_message  ," queue) File ""/""", line 605, in get_list  response = (action, params, path," verb) File ""/""", line 592," in make_request  headers=headers) File ""/""", line 461, in make_request  return (method, path, data, headers, host," sender) File ""/""", line 395," in _mexe  response = () File ""/""", line 1027," in getresponse  () File ""/""", line 407, in begin  version, status," reason = () File ""/""", line 365," in _read_status  line = () File ""/""", line 430, in readline  data = recv(1)\'',,,,,,,,,,
boto,564,open,1,2012-02-07 23:45:48,20.67,1,,,"b'b"" I believe the socket timeout is global", so worse case, you may be able to set it via the socket module. Worse case," we\'re working on a re-factor to replace a bunch of the plumbing that may resolve this.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,2,2012-02-07 23:45:48,885.28,,1,,b'b\' Well," if you had bought my handy ""Python and AWS Cookbook"" 8^) 8^) you would know that the socket time can be configured in your boto config file like this: [Boto]http_socket_timeout = 5 The timeout is specified in seconds.  @gtaylor is correct that this is a global setting that is actually buried down in the socket module.  This just gives you a convenient way to control it.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,3,2012-02-07 23:45:48,2.15,,1,,"b'b"" I\'m going to close this issue.  Please re-open if you feel some action should be taken in boto to address it.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,closed,2012-02-11 14:53:54,908.1,2,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,4,2012-02-07 23:45:48,245.93,1,,,"b'b"" @garnaat  Yes I think boto needs to take action to address this issue. All connection object should have a timeout.  This isn\'t just for boto",  from now on every time you write a connection object it should have a timeout.  Requireing us to go though every damn socket in our entire application and just to change the timeout is a fucking nightmare.  It was less code for us to modify boto,"  because that is where the mistake is. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,5,2012-02-07 23:45:48,1.48,1,,,"b""b' @gtaylor  Asking us to adjust the timeout for every other socket in the entire application is a really horrible suggestion.  We hacked up boto","  because every connection object needs a timeout. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,6,2012-02-07 23:45:48,2.43,1,,,"b""b' @MikeBrooks Blame Python"," not me. The global socket timeout is nothing new: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,7,2012-02-07 23:45:48,8.48,,1,,"b""b' @gtaylor Every connection object needs a timeout value.  Python adheres to this important design",  httplib and all of their connection objects have a timeout value that can be passed in to their constructor and has a  method. () defaults to None,  so boto crashes at 0% cpu for infinite time.  Yes,"  this is a very serious problem. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,8,2012-02-07 23:45:48,4.52,1,,,"b""b' @MikeBrooks Your anger is misdirected.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,9,2012-02-07 23:45:48,7.63,,1,,"b'b"" You can override it in the config file.  We can add an optional parameter to Connection objects.  I would rather make the change once in the config rather than have to remember everytime I create a Connection but it\'s not a difficult thing to add if it helps.  Note that the timeout parameter to httplib connections was not added until Python 2.6.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,10,2012-02-07 23:45:48,12.55,,,1,"b'b"" @garnaat This doesn\'t have to break compatibility for anyone.  Have a default value timeout=None (or check if its None and set it to ())"," the user can override it if they want but no one is forced to.  This is exactly the call convention for httilib. Having a  for this value would also be very nice. (I didn\'t even know there was a config file.)""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,11,2012-02-07 23:45:48,7.92,,,1,"b'b"" As I said"," I\'m fine adding that.  The only reason it\'s not there is that no one has asked for it before.  A previous discussion: url  resulted in a pull request that added the timeout to the boto config file.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,12,2012-02-07 23:45:48,9.33,1,,,"b""b' I think that both systems would work well together if the default was the config file value.  Cool",  sorry for being such a dick and a regularity-nazi. Thank you for hearing me out,"  we both want Boto to be a strong project. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,13,2012-02-07 23:45:48,12.57,,1,,"b""b' No problem"," sounded like it was all pretty fresh in your mind so you undoubtedly just got burned by it.  Totally understand.  i will re-open.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,reopened,2012-02-20 20:06:46,312.87,7,6,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,14,2012-02-07 23:45:48,310.07,,,1,"b""b' I know this is old"," but I too would love a connection-specific timeout '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,15,2012-02-07 23:45:48,1425.2,,1,,"b""b' Anyone still reaching this thread should see:url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,564,open,-,-,-,7,7,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,opened,2012-07-30 18:13:17,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,title,2012-07-30 18:13:17,00:00,,1,,"b""b' Changes to convert dynamoDb number type to a python decimal type"," to pre...'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,body,2012-07-30 18:13:17,00:00,,1,,"b""b' ...serve the 38 decimal digits that dynamoDb supports (after the decimal point). This avoids converting the number to a float or int"," which causes problems since the value no longer matches what is in dynamoDb. See boot issue 873 for more details.url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,1,2012-07-30 18:13:17,1120.83,1,,,"b'b"" I should have used Decimal to begin with.  That was my mistake.  I\'m trying to figure out the impact", if any, to existing users.  Any thoughts on that?  I\'ll do some local testing," too.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,2,2012-07-30 18:13:17,122.37,,,1,b'b\' Some thoughts from looking at the decimal docs : If existing users were using the results for arithmetic operations they might see TypeError exceptions (since the returned item is now a Decimal instead of an int or float). It looks like Decimal() + int() is fine, but Decimal() + float() gives an exception.  Casting the a Decimal to a float (or vice-versa) seems simple," although it is an extra step.  url I guess that if an application switched to using Decimal instead of floats it would require more memory.  For me that is a ""don\\\'t care"" and I guess for most people (i.e. using dynamoDB more for ""high-availability"" than for massive storage).  Actually for me it is a plus since we are storing long numeric keys and want to keep all the digits!  This is more of a ""future"" issue", than a backward compatibility issue, but I noticed this in the Decimal docs.  I was wondering if an application took in arbitrary Decimal values, and stored them in dynamoDB, what would happen for the special values below (i.e. what would dynamoDB do with them).  There is already a conversion when storing things in dynamoDB anyway (38 digits max after the decimal point, etc.) so you can\\\'t expect an arbitrary Decimal to come back unchanged but it would be nice to figure out the limits. 9.4.5.2. Special valuesThe number system for the decimal module provides special values including NaN, sNaN, -Infinity, Infinity, and two zeros," +0 and -0. My $0.02 is that Decimal is a good match to the large numbers allowed by DynamoDB and is the best default. For backward compatibility maybe there could be a configuration option to have the ""old behavior""", so if someone upgrades and see exceptions (that they don\\\'t have time to fix right away) they can set a config option and quickly get the old behavior until they have time to deal with the exceptions.   Regards,Eric\'',,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,3,2012-07-30 18:13:17,34.32,,,1,b'b\' I think I\\\'d prefer to extend Decimal to allow it to warn (or die) during truncation, handle  math," and implement handling for the special values.  Allowing the user to override our ""DynamoNumber"" class would allow for both lightweight (float?) use or more sophisticated implementations.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,4,2012-07-30 18:13:17,143.78,,,1,"b'b"" If there was a config option to use Decimal that would work for me also.  I haven\'t looked at the boto config options", but if it could be passed in  ( aws_access_key_id=zzz, aws_secret_access_key=zzz, dynamoNumber=\'Decimal\') it would be very convenient (one line in the code).  A separate config file would be less convenient," but would work too.  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,closed,2012-08-23 17:54:35,1421.3,1,2,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,reopened,2012-08-23 17:55:16,0.68,1,2,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,5,2012-07-30 18:13:17,0.83,1,,,b'b\' oops, I meant to comment," not ""close and comment"".  \''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,6,2012-07-30 18:13:17,127.07,,1,,b'b\' I\\\'ve made some changes locally and I\\\'m testing, but when I try to add a Decimal attribute to and Item and then store it, I\\\'m getting an error from DynamoDB. Gives me: DynamoDBValidationError: DynamoDBValidationError: 400 Bad Request{u\\\'message\\\': u\\\'Attempting to store more than 38 significant digits in a Number\\\'," u\\\'__type\\\':  u\\\'#ValidationException\\\'} This is what is actually sent over the wire:  {""Item"": {""foo"": {""S"": ""mitch""}","bar"": {""N"": ""3""}, ""fie"": {""N"": ""33.4560000000000030695446184836328029632568359375""}}, ""TableName"": ""foobar""}\''
boto	890	closed	7	elee-nst	2012-07-30 18:13:17	72.97			+1	b'b\' Here is my understanding:  From: url  By default Decimal will store the ""exact"" value (i.e. tries to not lose precision when doing conversions). There is a  getcontext().prec = 28setting to tell decimal how many decimal digits to store.  This is used when computing the results of an operation with decimals (, etc.). There is also a .quantize() method that will trim to the desired number of digits. Since amazon dynamodb supports 38 decimal digits of precision (after the decimal point), I tried setting the ().prec to 38.  But, that didn\\\'t give 38 digits after the decimal point, I needed to set it to 40 to give room for the integer portion also, ""33""*\\\\* THE QUANTIZE METHOD NOW TRIMS TO 38 DIGITS AFTER THE DECIMAL POINT.  MULTIPLYING BY ONE WILL ALSO CAUSE IT TO TRIM THE RESULTS TO THE \\\'PREC\\\' SPECIFIED.  One trick seems to how to specify 38 digits AFTER the decimal point (which is the dynamoDB limit).  But, this looks like it is the right place to tweak things in the decimal library. Rgds,Eric\''
boto	890	closed	8	elee-nst	2012-07-30 18:13:17	16.98			+1	b'b\' Or, perhaps it could be trimmed to 38 digits (after the decimal point) after it is converted to a dynamo string? def dynamize_value(val):  """"  Take a scalar Python value and return a dict consisting  of the Amazon DynamoDB type specification and the value that  needs to be sent to Amazon DynamoDB.  If the type of the value  is not supported, raise a TypeError  """"  def _str(val):    """"    DynamoDB stores booleans as numbers. True is 1, False is 0.    This function converts Python booleans into DynamoDB friendly    representation.    """"    if isinstance(val, bool):    return str(int(val))    return str(val) perhaps replacing str(val) with something like:  str(val)[ 0 : str(val).find(\\\'.\\\')+39 ]i.e. return everything from the start of the string to 38 digits after the . I\\\'m not certain that is 100% right, I think the str(val).find returns -1 if there isn\\\'t a decimal point, so a big positive integer might possibly be trimmed to 37 characters?  maybe something like the code below (warning, I didn\\\'t have time to test this runs!). string = str(val)hasDot = (\\\'.\\\')if hasDot   string = string[ 0 : hasDot+39 ]return string\''
boto	890	closed	9	garnaat	2012-07-30 18:13:17	975.4	+1			b'b"" I could trim the string prior to sending it to DynamoDB.  I\'m just wondering what would be better from a user perspective.  Trimming behind the scenes may be confusing because the value actually stored in DynamoDB would be different than the value the user sees in Python.  Not trimming would probably result in errors like this from DynamoDB but it would force the user to explicitly deal with it in Python.  I\'m kind of leaning towards the latter approach.  Comments?""'
boto	890	closed	10	disruptek	2012-07-30 18:13:17	163.55		+1		b'b"" Leaving the handling to the user seems like the right approach to me; it\'s more pythonic and doesn\'t obscure semantics that we can\'t automate expertly.  If we don\'t like my idea of a DynamoNumber class that can handle that expertise and be overridden by the user, how about designing the number handling code as abstractly as possible, so that non-Decimal instances can be consumed even if Decimal instances are produced?""'
boto	890	closed	11	garnaat	2012-07-30 18:13:17	12.85		+1		b'b"" @disruptek It\'s not that I don\'t like the idea of a DynamoNumber class, I just haven\'t had time to implement it.""'
boto	890	closed	12	elee-nst	2012-07-30 18:13:17	67.08			+1	b'b\' One thought (which I have not looked at much, so it might not be practical).  There might be cases (like 33.456) where a user is expecting dynamoDB to contain exactly 33.456, not the ""nearest floating point number"" of \\\'33.45600000000000306954461848363280296326\\\'. Since dynamoDB sends everything as a string when talking to the cloud, perhaps there could be an option to  out a number as a string (where the end user has full control over the number of digits, etc.). The only hiccup I could see is that dynamoDB will remove  0\\\'s (if you send 1.234000000 I believe it will store it as 1.234) so you could not always do an exact string match between what you sent and what you get back.  But that is inherent in dynamoDB, any library interfacing to is would have this behavior.  \''
boto	890	closed	13	disruptek	2012-07-30 18:13:17	152.85		+1		b'b\' I like that idea.  Maybe what we really want is for our internal encoding of numeric values to be strings.  This matches the DynamoDB model more explicitly and allows for the ""33.456"" string assignment without any lossy casting.  I\\\'d like to contribute some work on this, but I\\\'m pretty well slammed at the moment.\''
boto	890	closed	14	elee-nst	2012-07-30 18:13:17	866.37	+1			b'b\' My apologies for a slow response!  I am not that familiar with theLayer1/Layer2 model of dynamoDB (I am a new user) and did not have a chanceto read up on it until now.It does seem to me that Layer1 is a good place to allow json strings like""SomeNum"":{""N"":""1307654345.1234567890123456789""} to be directly input. url This is the lowest-level interface to DynamoDB. Methods at this layer mapdirectly to API requests and parameters to the methods are either simple,scalar values or they are the Python equivalent of the JSON input asdefined in the DynamoDB Developer\\xe2\\x80\\x99s Guide. All responses are directdecoding of the JSON response bodies to Python data structures via the jsonor simplejson modules. What might the API look like?  Do you have a small example showing how itwould be used? Eric On Tue, Sep 11, 2012 at 4:26 AM, jimdanz notifications@ wrote:##  Eric Lee408.505.8262 NOTICE: This email message is for the sole use of the intended recipient(s)and may contain confidential and privileged information. Any unauthorizedreview, use, disclosure or distribution is prohibited. If you are not theintended recipient, please contact the sender by reply email and destroyall copies of the original message.\''
boto	890	closed	15	disruptek	2012-07-30 18:13:17	713.1	+1			b'b"" I agree that Layer1 should match the API as closely as possible, but as with any protocol design, I think we want to be strict in what we produce and loose in what we consume.  I don\'t see the harm in performing Layer1 string conversions on numeric types that we find in lieu of strings, but any lossy precision conversion or -flow should raise an exception. We can perform the conversion to Decimal using a custom Context that asserts those conditions.  This will break existing code that is using lossy float storage, but it will expertly handle strings, (exact) floats, and numeric types including bool and Decimal. Decimal does seem to be the appropriate type for wrapping Layer2 numeric values retrieved from DynamoDB, as it will yield lossless reads.  The user can drop to Layer1 if they want to perform their own type conversions with an eye towards resource conservation or controlled loss of precision. Again, this almost certainly will yield new exceptions in existing code, but the reality is that the sooner we make a change, the better.""'
boto	890	closed	16	jamesls	2012-07-30 18:13:17	69.52		+1		b""b' Closing in favor of #1183, which supercedes this.'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,closed,2013-01-07 23:53:09,357.88,5,7,6,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,890,closed,-,-,-,5,7,6,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,opened,2012-10-14 03:23:00,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,title,2012-10-14 03:23:00,00:00,,1,,"b""b' Glacier code returning Name or service not known on big files'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,body,2012-10-14 03:23:00,00:00,1,,,"b'b\' In uploading a 10GB archive (tar file verified) it seems the connection method is failing. Traceback (most recent call last):  File ""glacier_upload_""", line 12," in   archive_id = (filename)  File ""/-""", line 77," in upload_archive  return (filename)  File ""/-""", line 140," in create_archive_from_file  (data)  File ""/-""", line 152," in write  ()  File ""/-""", line 141, in send_part  content_range," part)  File ""/-""", line 625," in upload_part  response_headers=response_headers)  File ""/-""", line 78," in make_request  data=data)  File ""/-""", line 910, in make_request  return (http_request, sender," override_num_retries)  File ""/-""", line 872," in _mexe  raise e: [Errno -2] Name or service not known I get this uploading from a large EC2 instance. My code is: import botoimport  import sys GLACIER_VAULT = \\\'dropbox-photos-vault\\\'ONE_MB = 1024 \\\\* 1024if **name** == ""**main**"":  \''",,,,,,,,,,,,,,,,,,,,
boto,1055,closed,closed,2012-10-14 03:34:24,11.4,1,1,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,reopened,2012-10-14 03:34:30,0.1,1,1,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,1,2012-10-14 03:23:00,538.57,,1,,b'b\' After further investigation, I believe that the connection pool associated with a particular (host, is_secure) key just simply continues to grow due to the value of (conn) in  always returning false. This continues to grow until we encounter an error. This is a serious issue for large uploads. The connection pooling in  needs improving. Firstly, there is no limit to the number of connections that can be in a single pool. Secondly, there is no closing of connections. I suspect that that code:  def _conn_ready(self," conn):    """"""    There is a nice state diagram at the top of .  It    indicates that once the response headers have been read (which    _mexe does before adding the connection to the pool)", a    response is attached to the connection, and it stays there    until it\\\'s done reading.  This isn\\\'t entirely true: even after    the client is done reading, the response may be closed, but    not removed from the connection yet. Needs improvement.  I will look for an elegant fix and prepare a pull request.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,closed,2012-10-17 17:27:31,833.02,1,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,2,2012-10-14 03:23:00,752.43,,1,,"b'b\' The fix does not seem to work -_conn_ready still always returns False and eventually everything crashes with ""Name or service not known"".\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,3,2012-10-14 03:23:00,754.72,,1,,"b""b' stigger"," can you post any code for me to look at?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,4,2012-10-14 03:23:00,761.62,,1,,"b'b"" I\'m using glacier-cmd url  It could be that the problem is not in boto", but in glacier-cmd itself, however," it just calls () (:1086).""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,5,2012-10-14 03:23:00,764.33,,1,,"b""b' Do you know what version of boto it is using? '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,6,2012-10-14 03:23:00,767.43,,1,,"b'b"" I just checked out the  branch and tested with it \\xe2\\x80\\x94 problem isstill there"," the () trick didn\'t fix it.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,7,2012-10-14 03:23:00,768.83,,1,,"b'b"" Can you post in the stack trace you\'re seeing"," and any other relevant information.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,8,2012-10-14 03:23:00,801.73,1,,,"b'b"" Apparently", I was wrong -they are calling layer1.upload_part directly. I applied that () trick there, and now it works as supposed to. I\'m very sorry for bumping this issue. boto works correctly," this issue is definitely fixed.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1055,closed,-,-,-,2,8,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,opened,2012-10-18 15:50:52,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,title,2012-10-18 15:50:52,00:00,,1,,"b""b' use ElementList instead of Element for ...'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,body,2012-10-18 15:50:52,00:00,1,,,"b""b' .. The GetReportRequestList response can contain multiple ReportRequestInfo elements"," and using Element for these results in the loss of all but final element.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,1,2012-10-18 15:50:52,97.15,,,1,"b""b' Thanks"," but this is already fixed in develop.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,2,2012-10-18 15:50:52,7.7,,,1,"b""b' Awesome"," thank you.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,closed,2012-10-18 17:35:44,104.87,1,1,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,3,2012-10-18 15:50:52,4.45,,1,,"b'b"" Ah", actually, I see that this has been fixed for in 293ae43a," but it doesn\'t look like the same fix was applied for  .""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,reopened,2012-10-18 17:40:10,4.43,1,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,4,2012-10-18 15:50:52,2.63,,,1,"b'b"" You\'re right; that somehow escaped my commit.  Thanks for the catch.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,closed,2012-10-19 15:05:46,1285.6,1,2,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1067,closed,-,-,-,1,2,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,opened,2012-12-01 19:21:38,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,title,2012-12-01 19:21:38,00:00,,1,,"b""b' Proposal: keyring support'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,body,2012-12-01 19:21:38,00:00,,,1,"b'b"" For greater security", I\'d really rather store my AWS -key in a secure keyring. I propose to add keyring url  to boto, as follows:In a boto configuration Credentials section, supply a aws_access_key_id and keyring options:  [Credentials]  aws_access_key_id = your_access_key  keyring = aws  If boto sees keyring, rather than aws_secret_access_key, then it will import keyring and call with the keyring name and the  aws_access_key_id. If you agree, I\'ll provide a pull request (with a test," of course) for your review.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,1,2012-12-01 19:21:38,242.77,,1,,"b""b' In the mean time: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,closed,2012-12-02 23:24:24,242.77,0,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,reopened,2012-12-02 23:24:29,0.08,0,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,2,2012-12-01 19:21:38,129.75,,,1,"b""b' I was not aware of keyring.  It looks very useful.  I would definitely be interested in a PR.  Thanks!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,3,2012-12-01 19:21:38,828.77,,1,,"b""b' Done: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,4,2012-12-01 19:21:38,153.42,,,1,"b""b' Merged in a35a8bc"," thanks.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,closed,2012-12-21 17:56:20,1111.85,0,3,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1154,closed,-,-,-,0,3,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,opened,2013-03-11 12:29:17,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,title,2013-03-11 12:29:17,00:00,,1,,"b""b' Route53 failover'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,body,2013-03-11 12:29:17,00:00,,,1,"b""b' Hi"," Will boto support failover on route53: url  It would be great if we could  records with health checks set. Thanks.Kind regards.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,1,2013-03-11 12:29:17,708.6,,1,,"b'b"" It definitely will support failover.  I can\'t give you an exact date but it\'s high on our todo list.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,2,2013-03-11 12:29:17,516.42,,,1,"b""b' Great ! Thank you for your efforts.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,closed,2013-03-12 08:54:18,1225.02,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,3,2013-03-11 12:29:17,599.08,1,,,"b""b' Sorry", about reopening but," is there any  news on this ? Thank so much.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,reopened,2013-07-03 18:53:23,599.08,1,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,4,2013-03-11 12:29:17,1012.83,,,1,"b""b' Yeap"," the same question!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,5,2013-03-11 12:29:17,1321.72,,,1,"b'b"" +1. I\'d love to be able to manipulate Route53 health checks.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,6,2013-03-11 12:29:17,35.0,,1,,"b""b' +1. However"," it seems to be implement in PR #2236Looking for documentation of this feature.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1378,open,-,-,-,1,3,4,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,opened,2013-04-30 13:21:23,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,title,2013-04-30 13:21:23,00:00,,1,,"b""b' CloudSearch complex ranking not supported'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,body,2013-04-30 13:21:23,00:00,,1,,"b'b"" When using complex query-time ranking", one is supposed to use the **rank_[identifier]_** parameter and then pass the _[identifier]_ into **rank**. This does not appear to be currently supported by boto. Note that simply passing the ranking query to **rank** does not work. Perhaps **rank** should be a dictionary of \'name\': \'query\'? Or, to allow all edge cases," maybe () should accept a **kwargs that is passed into the request pipeline.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,1,2013-04-30 13:21:23,106.23,,1,,"b""b' Does anyone have any thoughts on this? Am I the only one concerned about the lack of complex ranking support? If people agree on the kwargs idea"," I can fork and commit in short order.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,closed,2013-05-18 15:07:37,106.23,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,reopened,2013-05-18 15:09:50,2.22,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,closed,2013-05-18 15:09:55,0.08,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,reopened,2013-05-18 15:09:56,0.02,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,2,2013-04-30 13:21:23,2.9,1,,,"b'b\' I apologize for the ""reopens""", I keep hitting the wrong button.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,3,2013-04-30 13:21:23,220.37,,,1,"b'b"" I think the best solution would be to allow the _rank_ parameter to accept a dictionary", while preserving backwards compatibility. If you want to work on that, as well as make sure there are tests to preserve backwards compatibility," I\'m sure a pull request would be most welcome.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1466,open,-,-,-,1,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,opened,2013-06-11 01:35:07,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,title,2013-06-11 01:35:07,00:00,1,,,"b""b' Fix annoying typo in basic DynamoDB example'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,body,2013-06-11 01:35:07,00:00,,,1,"b'b"" This actually makes the example copy&paste \'able Thanks a lot!""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,closed,2013-06-11 01:36:58,1.85,1,0,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,reopened,2013-06-15 20:18:06,1121.13,1,0,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,1,2013-06-11 01:35:07,1123.33,1,,,"b""b' accidentally closed this :tired_face: '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,closed,2013-06-15 20:48:20,30.23,2,0,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,2,2013-06-11 01:35:07,30.05,,,1,"b""b' Thanks for the fix!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,3,2013-06-11 01:35:07,412.45,,,1,"b""b' Thanks for the fix merge!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1551,closed,-,-,-,2,0,3,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,opened,2013-06-25 10:57:28,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,title,2013-06-25 10:57:28,00:00,,1,,"b""b' SQS: SignatureDoesNotMatch error since 2.9.3'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,body,2013-06-25 10:57:28,00:00,,,1,"b'b"" I\'m using Celery with SQS and ever since 2.9.3 I\'ve been getting a SignatureDoesNotMatch error. Looking at the [diffs]url  I\'ve noticed there were changes to the auth signature. How can I best diagnose this"," because I\'ve not seen anyone else with a similar issue and 2.9.3\'s been out for a month!  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,1,2013-06-25 10:57:28,481.52,,1,,"b'b"" I\'m not able to reproduce this", both on 2.9.3 and on the latest version 2.9.6. Do you know what specific SQS operation is failing (reading from the queue, putting to the queue)? \\xc2\\xa0Or are all operations to SQS failing? If you turn on debug logging (there\'s a (\'boto\') helper function you can use) the debug logs will log what boto computed as the canonical string and string to sign. \\xc2\\xa0If these don\'t match up with what the error message from the server says it should be then we have a regression in boto\'s auth. \\xc2\\xa0Otherwise, it\'s most likely because the key used to sign the request is incorrect. James On June 25, 2013 at 3:57:49 AM, Michael Waterfall (notifications@) wrote:I\'m using Celery with SQS and ever since 2.9.3 I\'ve been getting a SignatureDoesNotMatch error. Looking at the diffs I\'ve noticed there were changes to the auth signature. How can I best diagnose this," because I\'ve not seen anyone else with a similar issue and 2.9.3\'s been out for a month!  SenderSignatureDoesNotMatchThe request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details. The Canonical String for this request should have been\'GET/Action=ListQueues&amp;Version=2012-11-05host:eu-west-1x-amz-date:20130618T110110Z host;x-amz-datee3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\' The String-to-Sign should have been\'AWS4-HMAC-SHA25620130618T110110Z20130618/eu-west-1/3c7019f0adae8b5fe064d24fd4d5a6d9ea732c4661f1263325765a69dd2c9fd6\'  1fc65a8f-5fa2-51d5-919a-10b51f95ab22.Trying again in 8.00 seconds...\\xe2\\x80\\x94\\xe2\\x80\\xa8Reply to this email directly or view it on GitHub.\\xef\\xbf\\xbc""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,2,2013-06-25 10:57:28,32.63,,1,,"b""b' This is related to #1552 and should be fixed as of SHA: 3f2b3386fee1f1ddb125d268080754f91cb765a9'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,3,2013-06-25 10:57:28,1086.82,,,1,"b'b"" Oh cool", thanks @toastdriven, glad to hear it\'s been addressed. @jamesls It was all operations that seemed to be failing, but looks like it\'s resolved in 2.9.6! Thanks for that info regarding debug logging," I\'ll make sure to look at that should I have any other issues. Cheers all!""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,closed,2013-06-26 13:38:27,160.98,0,3,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,4,2013-06-25 10:57:28,12.03,,1,,"b'b"" Nope", scratch that," I\'m still getting the error with 2.9.6. Looking into it now.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,reopened,2013-06-26 13:50:29,12.03,0,4,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,5,2013-06-25 10:57:28,16.75,,,1,"b'b"" With debug logging enabled"," this is the output for the list queues action. So I\'m guessing it\'s a regression? The keys are correct and work fine pre 2.9.3.  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,6,2013-06-25 10:57:28,18.17,,,1,"b'b"" Oh", also, have I gone an exposed anything secret that I shouldn\'t have? It\'s all just dev stuff," but I\'m just wondering because the seems to be consistent in all my logs! :-P""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,7,2013-06-25 10:57:28,237.5,,,1,"b'b"" Interesting", I\'m not able to reproduce this with the same region and operation and everything, but I\'ll dig in and see what I can find.  The is just the sha256 of an empty string," so there\'s no issues there:  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,8,2013-06-25 10:57:28,1264.83,,1,,"b'b"" This is still an issue for me and I can\'t upgrade past 2.9.2. @jamesls did you manage to find anything? Is there anything I can do to help locate the issue?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,9,2013-06-25 10:57:28,177.33,,,1,"b'b"" @jamesls I figured out the issue! I was specifying the port in my broker URL for celery", and that was making it\'s way into the Boto connection. The signature error happeness (and is reproducible) when port 80 is specified. It works fine if you specify  ," or 443. Is this a bug or expected behaviour?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1573,open,-,-,-,0,5,6,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,opened,2013-06-26 08:36:37,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,title,2013-06-26 08:36:37,00:00,,1,,"b""b' Propagate autoscale BlockDeviceMappings to launch configuration'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,1,2013-06-26 08:36:37,808.6,,,1,"b'b"" Thanks for the pull request. I\'m a bit confused about the purpose on this PR though. Can you explain what this is fixing or link to what\'s changed in the docs? We\'ll also need some tests to make sure we don\'t regress", as mentioned in the [contributing guide]url  If you could write some tests," that would help a lot. Otherwise this will have to wait until a core dev can find some time to write tests for this. Thanks again!""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,closed,2013-07-16 16:38:47,482.17,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,reopened,2013-07-16 16:38:51,0.07,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,2,2013-06-26 08:36:37,1120.2,,1,,"b'b"" My mistake... Thanks for the comments. I\'ll try to find some time to write the tests. The purpose of these patches was to make BlockDeviceMappings work", when used in Launch Configurations, as I remember it. It wasn\'t working for me when I tried, I think there was a mismatch in the arguments that get passed to the underlying API call. I ended up working around boto for this particular issue in my application. Anyway," tests should obviate what the problem really was. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1574,open,-,-,-,0,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,opened,2013-07-30 13:09:24,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,title,2013-07-30 13:09:24,00:00,,1,,"b""b' filters not recognised in any method'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,body,2013-07-30 13:09:24,00:00,1,,,"b'b\' In boto 2.9.9:import conn = (""us-east-1"")reservations = (filters={\\\'instance-state-name\\\': \\\'running\\\'}) Traceback (most recent call last):  File ""stdin""", line 1, in moduleTypeError: get_all_instances() got an unexpected keyword argument \\\'filters\\\'\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,1,2013-07-30 13:09:24,36.1,,1,,"b'b"" I\'m not sure what\'s going on here.  I can\'t reproduce this and the code for has not changed in a long time.  Are you saying that none of the methods will accept a param?  Could you verify your boto version by: Also"," which platform are you on and what version of Python are you using?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,2,2013-07-30 13:09:24,12.8,1,,,b'b\' I am on Ubuntu 10.04 LTS 64-bit.Here is the rest: Python 2.6.5 (r265:79063, Oct  1 2012," 22:04:36) [GCC 4.4.3] on linux2Type ""help""",copyright,"credits"" or ""license"" for more information.import botoboto.**version**Traceback (most recent call last):  File ""stdin", line 1, in moduleAttributeError: \\\'module\\\' object has no attribute \\\'**version**\\\' For the record, i can use other functions in boto like run_instances, terminate_instances, etc.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,closed,2013-07-30 13:58:18,48.9,2,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,reopened,2013-07-30 13:59:16,0.97,2,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,3,2013-07-30 13:09:24,9.58,,1,,"b""b' Ahh"," you must have been picking up that ancient version of boto that is bundled with Ubuntu 10.04.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,4,2013-07-30 13:09:24,29.48,,1,,"b""b' So the problem is ancient ubuntu? Because i surely tried to install the latest  it be fixed if i upgrade my system?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,5,2013-07-30 13:09:24,144.55,,1,,"b""b' You should still be able to install the latest boto.  You just need to makesure your PYTHONPATH is set up correctly so it finds the new one before theold one. On Tue", Jul 30, 2013 at 7:37 AM," eviekassela notifications@:'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,6,2013-07-30 13:09:24,99.6,,,1,"b'b"" From the beginning I run the commands in python2.6 shell.I made a script with #!/ on top. I also added PYTHONPATH in .bashrc. It still didn\'t work.Then I realized that maybe I should also reinstall boto. I used this command just to be sure: \'python2.6  install\' and after all it worked! Thank you for your immediate help.  p.s. In case this helps any future users", I couldn\'t also use add_tag and create_tags," except for filters.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,closed,2013-07-30 18:41:31,282.25,2,5,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1628,closed,-,-,-,2,5,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,opened,2013-08-11 16:21:46,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,title,2013-08-11 16:21:46,00:00,,1,,"b""b' [TypeError: string indices must be integers] while describing a dynamodb table'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,body,2013-08-11 16:21:46,00:00,,1,,"b""b' Brand new dynamodb table"," single primary hash key (String) created using AWS the web interface: Software:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,closed,2013-08-12 12:44:31,1222.75,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,reopened,2013-08-12 12:44:34,0.05,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,1,2013-08-11 16:21:46,253.2,,1,,"b""b' This is related to #1649.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,2,2013-08-11 16:21:46,108.13,,1,,"b""b' I created a new table in the AWS console and I just ran into this issue too: ..and when I try to use I end up with the same error.   output Any suggested work-arounds?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,3,2013-08-11 16:21:46,57.83,,1,,"b""b' Closed via #2161.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,closed,2014-03-14 23:20:56,636.37,0,5,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1663,closed,-,-,-,0,5,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,opened,2013-08-16 05:28:31,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,title,2013-08-16 05:28:31,00:00,,1,,"b'b"" Added logic to close files after they\'re uploaded to Glacier.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,body,2013-08-16 05:28:31,00:00,,1,,"b""b' Files that are opened by some of the Glacier methods are not being closed after they have been uploaded.  This patch should fix that.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,closed,2013-08-16 13:00:31,452.0,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,reopened,2013-08-16 13:01:44,1.22,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,1,2013-08-16 05:28:31,891.9,,1,,"b'b"" Can you provide more info on cases where Glacier methods don\'t close the file?  Specifically a repro script or a set of steps I can follow that shows when a file is not being closed?  I just want to make sure we\'re not missing any other scenarios here.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,2,2013-08-16 05:28:31,1178.52,,1,,"b""b' Hi. Basically", I found this bug when attempting to delete an archive afteruploading it with the multipart upload method from the layer2 glaciermodule. fails silently.  I surmise it is because that fileis still open. This happened on Windows. Please let me know if you need further details. On 8/19/13," James Saryerwinnie notifications@ wrote:'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1670,open,-,-,-,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,opened,2013-08-27 21:36:41,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,title,2013-08-27 21:36:41,00:00,,1,,"b""b' Basic GCM SNS Mobile Push Not Working'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,body,2013-08-27 21:36:41,00:00,,1,,"b'b\' I cannot seem to get a very basic use case of SNS Mobile Push using boto (2.10) to work.  When I use the exact same message payload on AWS\\\'s webapp for using SNS Mobile Push url  I see the expected message (""foo"") appear on my Android device.  However", when I use boto,"null"" appears on my Android device.  I\\\'ve tried lots of different flavors of messages (where the value of ""GCM"" is not a string, but more nested JSON structure), to no avail.  The documentation has no examples of GCM SNS Mobile Push, so I\\\'m flying blind. A simple app that demonstrates the issue appears here:  url The target_arn has its last few characters obscured so as not to invite hackers to send stuff to my device, but when the correct value is used, I _do_ receive push messages -just the wrong value.\''
boto	1692	closed	1	toastdriven	2013-08-27 21:36:41	28.12			+1	b'b"" Combined with SHA: 62fdf34e8791cdad3fc811e515b03623366bc264, if you change your code sample to url  it should work properly. I don\'t have an Android device to verify this with, so confirmation would be appreciated.""'
boto	1692	closed	closed	 - 	2013-08-28 22:04:49	28.13	0	2	1
boto	1692	closed	2	eswenson1	2013-08-27 21:36:41	21.9		+1		b'b\' Thanks. But this still results in ""null"" being received as the value of ""message"" on the GCM device (Android Galaxy S3).  I pulled your latest code (and see your fix to publish), updated my test case to supply the new message_structure=\\\'json\\\' parameter to publish, but with the same results.  I then simply used your pastebin and changed the ARN to be the ARN for one of my devices, and I still see ""null"". I wrote a simply script to invoke GCM directly and, with the same payload, am able to see the correct message on my device.  So it must be something with your wire interface to AWS SNS that doesn\\\'t meet its expectations.\''
boto	1692	closed	reopened	 - 	2013-08-28 23:26:40	81.85	0	3	1
boto	1692	closed	3	eswenson1	2013-08-27 21:36:41	78.83			+1	b'b"" By the way, I used the SNSMobilePush Java application (comes with the aws java sdk) and changed it to send the same payload.  It works fine and I can see correct messages showing up on the device.  You might consider looking at the HTTP requests generated by that application and comparing with the requests generated by  to see what is different.  I was about to do that but got dragged into a completely unrelated firestorm and haven\'t gotten to it.""'
boto	1692	closed	4	toastdriven	2013-08-27 21:36:41	1423.45	+1			b'b"" I spent much of the day on this today. I haven\'t been able to get a mobile app receiving notifications yet (because mobile is hard), but after comparing what the Java client sends, url  produces a very close body. Would you mind giving that a try & see if it works for you? I\'m going to keep trying to get a working end-to-end test by hand as well.""'
boto	1692	closed	5	eswenson1	2013-08-27 21:36:41	66.98	+1			b'b\' Hi Daniel, Do I need to pull a new version of boto from github?  When I run yoursns_mobile_, I get the following at the tail of the output: 2013-09-05 17:30:52,412 boto [DEBUG]:wrapping ssl socket; CA certificatefile=/2.10.0-py2.72013-09-05 17:30:52,600 boto [DEBUG]:validating server certificate:hostname=-east-1, certificate hosts=[u\\\'-east-1\\\']2013-09-05 17:30:52,693 boto[DEBUG]:{""Error"":{""Code"":""InvalidClientTokenId","Message"":""Client is notrecognized","Type"":""Sender""},""RequestId"":""e4a048fd-6424-557c-bb7f-025e405878af""}2013-09-05 17:30:52,694 boto [ERROR]:403 Forbidden2013-09-05 17:30:52,694 boto[ERROR]:{""Error"":{""Code"":""InvalidClientTokenId","Message"":""Client is notrecognized","Type"":""Sender""},""RequestId"":""e4a048fd-6424-557c-bb7f-025e405878af""}Traceback (most recent call last):  File ""sns_mobile_", line 31," in  target_arn=\\\'arn:*:arn:aws:sns:us-east-1:731381685129:-99c1-3365-902d-07d63f75ec7f\\\'  File""/2.10.0-py2.7""",line 255, in publish  return (\\\'Publish\\\'," params)  File""/2.10.0-py2.7""",line 726, in _make_request  raise (, , body): BotoServerError: 403 ForbiddenNone What am I doing wrong? (I\\\'m still using the copy of boto that I got afteryou integrated a a fix to add the message_structure parameter). -Eric On Thu, Sep 5, 2013 at 4:29 PM, Daniel Lindsley notifications@:\'',,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,6,2013-08-27 21:36:41,5.07,1,,,"b'b\' I tried after updating and instaling with the latest ""develop"" branch fromyour git repo and get the same errors: 2013-09-05 17:39:48",112 boto [DEBUG]:wrapping ssl socket; CA certificatefile=/2.12.0-py2.72013-09-05 17:39:48,384 boto [DEBUG]:validating server certificate:hostname=-east-1, certificate hosts=[u\\\'-east-1\\\']2013-09-05 17:39:48,"503 boto[DEBUG]:{""Error"":{""Code"":""InvalidClientTokenId""","Message"":""Client is notrecognized","Type"":""Sender""},""RequestId"":""bd88057c-a39e-5071-b769-4fdf850aebcb""}2013-09-05 17:39:48,503 boto [ERROR]:403 Forbidden2013-09-05 17:39:48,503 boto[ERROR]:{""Error"":{""Code"":""InvalidClientTokenId","Message"":""Client is notrecognized","Type"":""Sender""},""RequestId"":""bd88057c-a39e-5071-b769-4fdf850aebcb""}Traceback (most recent call last):  File ""sns_mobile_", line 31," in  target_arn=\\\'arn:*:arn:aws:sns:us-east-1:731381685129:-99c1-3365-902d-07d63f75ec7f\\\'  File""/2.12.0-py2.7""",line 255, in publish  return (\\\'Publish\\\', params, \\\'/\\\'," \\\'POST\\\')  File""/2.12.0-py2.7""",line 726, in _make_request  raise (, , body): BotoServerError: 403 ForbiddenNone On Thu, Sep 5, 2013 at 5:35 PM, Eric Swenson eric@ wrote:\'',,,,,,,,,,,,,,,,,,,
boto,1692,closed,7,2013-08-27 21:36:41,447.7,1,,,"b""b' Odd. The in the traceback sounds like a configuration error or something. Is that token still setup in SNS? Does sending the same data work with the Java client right now?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,8,2013-08-27 21:36:41,523.43,,1,,"b""b' For comparison"," this is what Java generates over the wire: And this is what Boto generates with the provided Gist:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,9,2013-08-27 21:36:41,396.32,,1,,"b'b"" Closing for now", as we\'re basically identical to the Java request. If a further test case can be worked up to show where this falls down," please reopen.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,closed,2013-09-12 23:28:30,1.83,4,5,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,10,2013-08-27 21:36:41,61.63,,,1,"b'b"" Thanks", Daniel.  I\'ll give it a try when I have a chance and let you know. I appreciate your help. On Thu, Sep 12, 2013 at 4:28 PM," Daniel Lindsleynotifications@:""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,11,2013-08-27 21:36:41,11.65,1,,,"b'b"" Regarding sending push message to android using SNS Publish I can\'t get the AWS CLI (based on boto) or Paws (also based on boto) to send the message payload no matter how I format it.I get the notification but it is empty.Using the AWS console I get the message.if I execute this aws cli for a working target-arn it works for iOS and only the icon shows for android.I\'ve tried using Paws with perl and I get the same result.I have a java implementation that seems to work but I\'d have to confirm it is with the same parameters.Has anyone gotten an SNS message through to an android client using boto?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,12,2013-08-27 21:36:41,468.88,,,1,"b""b' Its 2018 and its still not working... Amazon really needs to fix it in SNS.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1692,closed,-,-,-,5,5,4,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,opened,2013-09-30 10:08:03,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,title,2013-09-30 10:08:03,00:00,,1,,"b""b' Add timeout option to SSH client.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,body,2013-09-30 10:08:03,00:00,,1,,"b'b"" I\'m not sure where to test this option.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,1,2013-09-30 10:08:03,643.13,,,1,"b'b"" This looks generally good. I\'ll try to add a test and merge it in.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,closed,2013-10-14 18:40:22,512.32,0,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,reopened,2013-10-14 18:41:45,1.38,0,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,2,2013-09-30 10:08:03,160.97,,1,,"b""b' Fixed in 653b82b9c14241402bc81a77278895627aefb460.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,closed,2013-10-14 23:32:09,290.4,0,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1755,closed,-,-,-,0,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,opened,2013-10-11 21:08:37,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,title,2013-10-11 21:08:37,00:00,,1,,"b""b' Adding support for QueueOwnerAWSAccountId in GetQueueUrl to SQS'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,body,2013-10-11 21:08:37,00:00,,1,,"b""b' Support for optional parameter found here: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,1,2013-10-11 21:08:37,1249.65,,1,,"b""b' Closing as duplicate of #1786.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,closed,2013-10-14 17:58:16,1249.65,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,2,2013-10-11 21:08:37,18.15,1,,,"b'b"" Whoops", sorry about that," I didn\'t see that this is for the branch. Reopening.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,reopened,2013-10-14 18:16:25,18.15,1,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,closed,2013-10-14 18:36:06,19.68,1,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1787,closed,-,-,-,1,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,opened,2013-11-04 20:39:41,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,title,2013-11-04 20:39:41,00:00,,1,,"b""b' RDS "," SqlServer c2 audit mode'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,body,2013-11-04 20:39:41,00:00,,1,,b'b\' Hello," im having issue to retrieve \\\'c2 audit mode\\\' parameter value for SQL param groups it always returns ""False""Please advise.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,1,2013-11-04 20:39:41,217.38,,,1,"b""b' @yserinov please provide a code example that describes what you are trying to do"," and provide the output that you expect vs. the output you are getting. Thanks!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,2,2013-11-04 20:39:41,869.85,1,,,b'b\' Sorry for the hassle please close this issue," for some reason get value used as ""_value"" in MSSQL vs Oracle engine\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,closed,2013-12-20 20:04:21,1404.67,1,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,3,2013-11-04 20:39:41,261.03,,1,,"b'b"" I have the same issue when using 2.27.0. It appears to be related to the way boolean values are read and set from the parameter response in . In In SQL Server", parameters like \'c2 audit mode\' return \'1\' to indicate True. As such,the true value \'1\' becomes False after setup by boto. A suggested fix is to change the interpretation of truth (For now," I\'ve worked around this with .)""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,reopened,2014-03-18 20:02:12,1437.85,1,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,4,2013-11-04 20:39:41,55.1,,,1,"b""b' This does look like a legitimate issue. Please feel free to create a pull request with the fix and include tests to get it merged in quickly. Thanks!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1840,open,-,-,-,1,3,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,opened,2013-12-17 17:14:34,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,title,2013-12-17 17:14:34,00:00,,1,,"b""b' Switch to new-style classes'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,body,2013-12-17 17:14:34,00:00,,1,,"b""b' Most classes in boto are old-style", meaning that super calls are implemented using (self), instead of the more modern super(Class, self).method(). Old-style classes are deprecated and no longer supported in Python 3," and are always surprising to experienced Python programmers.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,1,2013-12-17 17:14:34,107.67,,,1,"b'b"" I\'ve been meaning to do this for months & just haven\'t gotten there"," but I\'d love to bite this off. Assigning to me.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,2,2013-12-17 17:14:34,364.17,,1,,"b'b"" The following is a list of places we use old-style classes: I have a branch started to convert these. Once I have these changed & tests are passing"," I\'ll submit a PR & link it up here.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,closed,2013-12-19 00:23:02,428.47,0,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,3,2013-12-17 17:14:34,17.98,,1,,"b'b"" You might also want to convert old-style super calls to new-style ones. I can open another issue for that if you\'d like.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,reopened,2013-12-19 18:49:36,1106.57,0,4,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,4,2013-12-17 17:14:34,1331.38,,1,,"b""b' Slightly less trivial to cover all of those. The list of places:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,5,2013-12-17 17:14:34,317.45,1,,,"b'b"" Yikes"," I\'d hate to see the awk command that generated that :)""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,6,2013-12-17 17:14:34,138.75,,1,,"b""b' No  "," just a little Python. :grinning: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,closed,2013-12-23 21:49:19,179.72,1,6,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,1918,closed,-,-,-,1,6,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,opened,2014-02-27 01:42:46,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,title,2014-02-27 01:42:46,00:00,,1,,"b'b\' Empty item is not ""falsey""?\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,body,2014-02-27 01:42:46,00:00,,1,,"b'b"" On the [Web documentation]url  I read:However", I don\'t seem to be seeing that behavior...  For example: I can work around this easily," but wonder if this is a bug...""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,1,2014-02-27 01:42:46,1284.82,1,,,"b'b"" I\'m afraid I don\'t quite follow. When a key isn\'t found", DDB returns an empty response ( ). Boto\'s sees this & should raise an exception url  What does return? If it\'s an instance, what\'s in  ? Also," what version of Boto are you using? Thanks.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,2,2014-02-27 01:42:46,164.43,,,1,"b'b"" Yes the return value of that command is an instance", and is  .  My question is whether should evaluate to  ," as seems to be suggested by the documentation. I\'m using version 2.9.6.  Thanks.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,closed,2014-03-15 01:52:03,9.28,1,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,reopened,2014-03-15 01:52:08,0.08,1,2,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,3,2014-02-27 01:42:46,109.07,,1,,"b'b\' The current version of Boto is 2.27.0. The ""false-y"" behavior was added in 2.20.0 url  so I\\\'d recommend upgrading if possible.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,closed,2014-03-15 03:41:05,108.95,1,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,4,2014-02-27 01:42:46,952.13,1,,,"b""b' Ah", right, I should have checked the versions.  Sorry! I had installed it on my system using  ," rather than  .'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2127,closed,-,-,-,2,3,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,opened,2014-02-27 02:39:38,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,title,2014-02-27 02:39:38,00:00,,1,,"b""b' missing methods in SES'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,body,2014-02-27 02:39:38,00:00,,,1,"b'b"" The following Amazon SES API methods don\'t exist in boto ses GetIdentityNotificationAttributes~~SetIdentityFeedbackForwardingEnabled~~~~SetIdentityNotificationTopic~~""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,closed,2014-03-10 20:35:08,1075.5,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,1,2014-02-27 02:39:38,1112.83,1,,,"b'b"" Thanks for the merge of my pull-request"," I didn\'t implement the GetIdentityNotificationAttributes method. Sorry for the possible confusion here but this issue should stay open until either myself or someone else can satisfy the remaining method""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,2,2014-02-27 02:39:38,13.05,1,,,"b""b' Reopening and editing the original text to reflect this. Sorry about that.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,reopened,2014-03-17 21:25:31,50.38,2,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2128,open,-,-,-,2,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,opened,2014-05-30 17:00:22,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,title,2014-05-30 17:00:22,00:00,,1,,"b""b' CloudSearch2 commit failure due to HttpAdapter max_retries. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,body,2014-05-30 17:00:22,00:00,,1,,"b'b"" in  on line 185", an HTTPAdapter is initialized, but the max_retries parameter doesn\'t exist in the constructor. This causes an exception. In ," the below is what works:  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,closed,2014-05-30 17:14:02,13.67,0,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,1,2014-05-30 17:00:22,219.6,,1,,"b""b' Is this still an issue?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,2,2014-05-30 17:00:22,12.58,1,,,"b'b"" Yes", I apologize if I didn\'t go through the correct channels. The problem described in the ticket is valid for 2.29.0, and the development version didn\'t work as well. It appeared someone closed the ticket without comment likely because I broke the boto process," so I created a fork in order to reference the fix I made (which is trivial). ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,reopened,2014-05-31 00:35:57,441.92,1,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,3,2014-05-30 17:00:22,1412.88,1,,,"b'b"" I\'m a bit confused. url  _does_ have the kwarg param. What version of requests are you using?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,4,2014-05-30 17:00:22,13.98,1,,,"b""b' I am sorry"," you are correct. I had an outdated version of requests. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,closed,2014-06-12 20:39:25,1203.47,3,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2300,closed,-,-,-,3,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,opened,2014-10-31 14:15:55,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,title,2014-10-31 14:15:55,00:00,,1,,"b""b'  #2296 Adding support for cloud formation UsePreviousValue parameter.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,body,2014-10-31 14:15:55,00:00,,1,,"b""b' In response to issue #2296.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,1,2014-10-31 14:15:55,436.13,1,,,"b""b' @danielgtaylor  Hi"," sorry to single you out. I was looking for someone with merge privileges to have a look at my PR. The linked issue #2296 is starting to get some interest.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,2,2014-10-31 14:15:55,369.22,,1,,"b""b' @aronkyle-lightspeedretail this needs to be updated and rebased.  hope it makes it in.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,3,2014-10-31 14:15:55,512.57,1,,,"b'b"" @joekiller I rebased the day before. Travis CI indicates that the tests work for python version 2.6 to 3.3. However"," it looks like the 3.4 tests are stalling in the cloudsearch section of the suite. This appears to be the case for all PR\'s up for the past 5 days or so. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,closed,2015-03-12 14:17:37,1.7,2,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,reopened,2015-03-12 14:17:43,0.1,2,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,closed,2015-03-12 14:19:24,1.68,2,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2725,closed,-,-,-,2,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,opened,2014-11-05 14:49:41,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,title,2014-11-05 14:49:41,00:00,,1,,"b""b' DynamoDB Document Data Types'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,body,2014-11-05 14:49:41,00:00,,1,,"b""b' I cant seem to find if supports it", and if it does it does not seem to be documented or I just cant find it. If i try to add with: I get: and with: I get:  I have seen some tickets and stackoverflow questions, but they all seem to lead to assumption that boto does not support it yet,"'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,1,2014-11-05 14:49:41,384.82,,1,,"b'b"" Don\'t know about the second part", but I convert my lists to sets prior to put. doc = {\\xe2\\x80\\x98first\\xe2\\x80\\x99: 1,\\xe2\\x80\\x98second\\xe2\\x80\\x99: 2,\\xe2\\x80\\x98items\\xe2\\x80\\x99: set([{\\xe2\\x80\\x98item_id\\xe2\\x80\\x99: 1, \\xe2\\x80\\x98foo\\xe2\\x80\\x99: 1, \\xe2\\x80\\x98bar\\xe2\\x80\\x99: 0},    {\'item_id\': 2, \\xe2\\x80\\x98foo\\xe2\\x80\\x99: 100," \\xe2\\x80\\x98bar\\xe2\\x80\\x99: 20}    ])}""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,2,2014-11-05 14:49:41,704.07,,1,,"b""b' @kmcintyre:  '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,closed,2014-11-10 08:58:34,1088.88,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,reopened,2014-11-10 08:58:41,0.12,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,3,2014-11-05 14:49:41,850.82,,1,,b'b\' Their example is a bit different than the code you\\\'ve submitted.  Notice how they use a table not a con," I can\\\'t really tell what ""con"" is in your example.  But list is definitely not supported", but set definitely is.\'',,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,4,2014-11-05 14:49:41,668.25,,1,,"b""b' Looks like document (Recursive types) are currently only supported by the Layer 1 API of Boto DynamoDB. Layer2 API does not yet support these.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,5,2014-11-05 14:49:41,600.17,,1,,"b'b"" It\'s a fairly small patch to Dynamizer to support :  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,6,2014-11-05 14:49:41,60.73,1,,,"b'b"" I\'m having trouble", with this patch I get a crc checksum error: This is when calling something like where the table already exists and mytable is an instance of  . Any ideas? I\'m connecting to a dynamo local instance (the latest version link of dynamodb local, just downloaded today) and haven\'t tried yet against real dynamo so I\'ll check now if that\'s where the problem is. UPDATE: This bug does not occur against the real dynamodb api," I can only recreate it with Dynamodb Local.  For the record I posted this bug the aws forum url ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,7,2014-11-05 14:49:41,12.15,,,1,"b'b"" @dcosson", I haven\'t tested against dynamodb local.  Real dynamodb seems to work fine for our use case.  It certainly hasn\'t been tested extensively," but  work well.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,8,2014-11-05 14:49:41,94.23,,1,,"b""b' @glinscott yeah it is a bug with dynamo db local"," works well for me against real dynamodb'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,9,2014-11-05 14:49:41,1415.62,,1,,"b'b"" What\'s the expected datatype on the resulting field?  Is it converted to a string set?  ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,10,2014-11-05 14:49:41,1279.73,,1,,"b'b"" @kmcintyre: no", dynamodb now supports \'document\' types.  so you can have  maps, and pure lists," with duplicates.  Basically a JSON document.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,11,2014-11-05 14:49:41,99.17,1,,,"b'b"" FYI url  This pull request has been in for quite some time now but no action has been taken yet. Based on the comments"," it\'s hard to say if this feature will ever make it in.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,12,2014-11-05 14:49:41,1149.32,,1,,"b'b"" @xethorn why wouldn\'t it make it? not support all the data types that dynamodb supports feels kinda limiting. ""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,13,2014-11-05 14:49:41,7.52,,1,,"b""b' @jrydberg : Unsure. I think the owner of the repo has some concerns about retrocompatibility. Feel free to chime in the pull request. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,14,2014-11-05 14:49:41,241.62,,1,,"b""b' Looks like this is fixed in boto v2.35.2.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,15,2014-11-05 14:49:41,3.22,,1,,"b""b' Yes \\xe2\\x80\\x93\\xc2\\xa0this issue should be marked as fixed.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,16,2014-11-05 14:49:41,0.48,,,1,"b""b' @Katafalkas @danielgtaylor"," feel free to close. Thanks for merging the PR!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,closed,2015-08-19 19:28:55,630.23,2,14,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,17,2014-11-05 14:49:41,1115.78,,1,,"b""b' Do we have any document or example for put item of Map type into dyanmodb using boto?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,18,2014-11-05 14:49:41,1152.77,,1,,"b""b' Highly recommend to switch to Boto3: url  If you cannot:  url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2737,closed,-,-,-,2,16,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,opened,2014-12-18 16:54:33,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,title,2014-12-18 16:54:33,00:00,,1,,"b'b"" Boto cannot find S3 objects where the name begins with a forward slash (\'/\')""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,body,2014-12-18 16:54:33,00:00,,1,,"b'b"" This might be a dupe of a previous issue: url  Having your application encode the URL prior to storing objects in S3 is a good workaround", but it won\'t work in cases where Boto needs to read an object written by another application that happens to begin with a forward slash. I looked at the debug output of for keys that do begin and do not begin with  , and found that the URI sent with the HEAD in both cases begins with only a single  . Somewhere," the extra is getting dropped before the request is sent to S3.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,1,2014-12-18 16:54:33,47.4,,1,,"b'b"" Also", if you use with a path that is prefixed by one or more forward slashes, they are removed. This is easy to avoid, but if you\'re working with objects placed by another application," then calling on the result of a can have surprising results.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,2,2014-12-18 16:54:33,34.88,,1,,"b""b' Looks like the answer is to use  . I found this in url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,closed,2014-12-18 18:16:50,82.28,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,3,2014-12-18 16:54:33,2.85,,1,,"b""b' Actually"," should this issue still be open? It seems like failing to operate properly on the result of a could be considered a bug even with the work-around.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,reopened,2014-12-18 18:19:41,2.85,0,5,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,4,2014-12-18 16:54:33,904.42,,1,,"b'b"" I am seeing this behavior also with some misapplied log files created right within the AWS console (ie log prefix started with /). (Listing a bucket and asynchronously storing the names to a queue for later processing.) However", turning on conn = (suppress_consec_slashes=False) worked for me as a workaround, at least for what I\'m currently working with. I think it could still be considered a bug (since output of  should be correct), but from the other things Mitch has said," this sounds like a really risky one to actually try to fix.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2824,open,-,-,-,0,6,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,opened,2015-02-04 20:52:37,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,title,2015-02-04 20:52:37,00:00,,1,,"b""b' CORS add_rule()","  error preventing allowed_header check.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,body,2015-02-04 20:52:37,00:00,,1,,"b'b\' S3 CORS clearly isn\\\'t being used very much. Simple copy paste error in  ""add_rule()"" when passing a single ""allowed header"" as a string. allowed_origin variable was incorrectly being used twice when the second type check should have been allowed_header (similar to expose_header check). All other tags are handled correctly.\''",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,1,2015-02-04 20:52:37,217.13,,,1,"b'b"" Thanks for the pull request! Seems we don\'t have a test covering this use case. Would you mind adding a unit test for this?""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,2,2015-02-04 20:52:37,94.77,,,1,"b""b' @danielgtaylor just added new test to cover this issue. Thanks!'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,closed,2015-03-03 16:26:47,1174.17,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,reopened,2015-03-03 16:27:00,0.22,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,2933,open,-,-,-,0,2,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,opened,2015-06-03 19:17:05,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,title,2015-06-03 19:17:05,00:00,,1,,"b""b' register_task_definition is missing the ability to define volumes.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,body,2015-06-03 19:17:05,00:00,,1,,"b""b'   takes only two arguments", family and container_definitions," however the actual API call has another argument of volumes. There is no way to currently use the boto with ECS. url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,1,2015-06-03 19:17:05,245.18,,1,,"b""b' We are de-prioritizing service updates to boto 2.x when they are available in boto3"," which can be used in conjunction with boto 2.x if necessary. [Here]url  that operation in the boto3 docs.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,closed,2016-03-28 23:22:16,245.18,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,2,2015-06-03 19:17:05,322.7,,1,,"b""b' What if I were to step up to the plate and maintain boto instead?'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,3,2015-06-03 19:17:05,1122.67,,1,,"b'b"" @gholms My wording was far to strict", for which I apologize. I edited my comment (and any similar comments on other issues) to emphasize that we are de-prioritizing service updates when those updates are available in boto3. We\'re still open to updates and Pull Requests, with the caveat that we believe the python community is best served, going forward, by putting our main effort into boto3 (there is some discussion on this [here]url  and in other place). We haven\'t done as great a job reviewing pull requests and issues as we should have," and we\'re working to improve that.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,reopened,2016-03-29 23:27:38,5.37,0,5,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,4,2015-06-03 19:17:05,12.58,,,1,"b""b' My team still has a substantial boto-using code base.  As much as we might like to port everything to boto3", that takes time.  It is quite difficult to use both the boto and boto3 interfaces for the same service at once because we wind up writing adapters for everything," so sometimes it just makes more sense to submit patches to boto.  If you need assistance reviewing pull requests I am happy to help with that.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3210,open,-,-,-,0,5,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,opened,2015-07-09 03:01:24,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,title,2015-07-09 03:01:24,00:00,,1,,"b""b' Problem importing boto in Python 3 running under Google Cloud'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,body,2015-07-09 03:01:24,00:00,1,,,"b'b"" Importing the module fails with the following message: After changing the line to  I got the error: So I simply imported urllib3 as urllib2: This seemed to solve the problem and I am now able to import boto. I haven\'t fully tested it to verify everything is working though (although I don\'t see why it shouldn\'t). For the record", my Python version is 3.4.2 and boto version is 2.38.0 Regards," Sam""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,1,2015-07-09 03:01:24,760.82,,1,,"b""b' In Python 3.x", using is required, hencewhy you got the error at the line. However, in Python 2.6+, the is optional.  In the , section, this exists: Thus, this is one of those things that needs to be updated for Py3 compatibility I would guess. The same goes for from its docs: You likely have installed because you have Python 3, while boto is coded for for Python 2.6 and 2.7.  TLDR: Boto is not 100% Python 3 ready yet," so YMMV. '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,closed,2015-07-15 01:27:42,1346.3,1,2,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,2,2015-07-09 03:01:24,433.22,,1,,"b'b"" Reopening"," this is causing py3 failures on travis.  We need to look into this and figure out what\'s going on.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,reopened,2015-12-21 22:55:26,1287.73,1,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,3,2015-07-09 03:01:24,1.68,,1,,"b""b' This is also being investigated here: url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,4,2015-07-09 03:01:24,1.93,,,1,"b'b"" Ahh", that explains why google\'s in the stack trace.  Interesting.  The code that\'s failing is not part of boto," it\'s a plugin that boto\'s loading at runtime. I\'ll need to see what we can do about that.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,5,2015-07-09 03:01:24,50.7,,1,,"b'b"" The problem boils down to the fact that the travis CI machines have a global file that is automatically being loaded.  One of the plugins called out in the file is not compatible with python3. Given we don\'t actually want to load a boto config file"," the easiest work around is to just set  .  I\'ve commented on the tracking issue on travis that the ideal solution is to a) get that plugin compatible with python3 or b) remove that line from  .  I\'ll keep tracking that issue but I think for now I\'d like to go ahead and just set this env var to get our builds passing in boto.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,6,2015-07-09 03:01:24,118.45,,1,,"b""b' Fixed via url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,closed,2015-12-22 01:48:12,172.77,1,6,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3263,closed,-,-,-,1,6,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,opened,2017-02-24 06:32:09,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,title,2017-02-24 06:32:09,00:00,,1,,"b""b' sqs region related signature fixed.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,body,2017-02-24 06:32:09,00:00,,1,,"b""b' fixed issue for sqs '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,1,2017-02-24 06:32:09,653.48,,1,,"b""b' better to use the current sqs endpoint. here url '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,closed,2017-02-24 17:25:38,653.48,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,reopened,2017-02-24 18:13:51,48.22,0,3,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,2,2017-02-24 06:32:09,48.47,,1,,"b""b' no additional test update needed with this commit '""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,closed,2018-05-30 03:54:13,580.37,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3688,closed,-,-,-,0,4,0,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,opened,2017-07-31 11:19:54,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,title,2017-07-31 11:19:54,00:00,,1,,"b""b' ec2/ add filter parameter to trim_snapshots()'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,body,2017-07-31 11:19:54,00:00,,,1,"b""b' Rebased url  (I no longer have access to the kixeye organization). This patch allows you to filter the trimmed snapshots. In some cases you may not want to trim all snapshots in an account or you may not have the same retention requirements for all snapshots. With this patch you can filter the snapshots to be trimmed by a tag.  If you require any changes before merging"," let me know.  Thanks.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,closed,2017-08-14 18:59:15,459.35,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,reopened,2017-08-14 18:59:19,0.07,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,closed,2017-08-17 09:41:37,882.3,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,reopened,2017-08-17 09:41:41,0.07,0,1,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,1,2017-07-31 11:19:54,1051.45,,,1,"b""b' @kyleknap @jamesls @danielgtaylor this was previously approved as #2377 but I had to create a new PR when rebasing.  Please advise if anything further is required.  Thanks.'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,2,2017-07-31 11:19:54,733.68,,1,,"b""b' @mfschwartz trying to find an active maintainer...'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,3,2017-07-31 11:19:54,502.32,,1,,"b'b"" @steve-dave",  I haven\'t worked with the ec2 snapshot code," so I\'d prefer for you to find a maintainer who has.  Mike""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,4,2017-07-31 11:19:54,746.93,,1,,"b'b"" @mfschwartz can you recommend an active maintainer that has worked with the code please?  You can see this has been open for a long time", I had it approved years ago (started off as #2377) and added tests, rebased," etc. and I\'m struggling to get anyone\'s attention...""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,5,2017-07-31 11:19:54,480.53,,1,,"b""b' @kyleknap @jamesls @danielgtaylor @mfschwartz ping'""",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,6,2017-07-31 11:19:54,504.63,1,,,"b'b"" @steve-dave in response to your question from Feb 8", I haven\'t been very involved with the boto community for some time," so unfortunately I don\'t have suggestions of a committer you could ask to  this change. Sorry.""'",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
boto,3746,open,-,-,-,1,5,2,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
